keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/ClusterDependencyContainer.kt

```diff
@@ -0,0 +1,74 @@
+package com.netflix.spinnaker.keel.api.ec2
+
+import com.netflix.spinnaker.keel.api.Locatable
+import com.netflix.spinnaker.keel.api.ResourceSpec
+import com.netflix.spinnaker.keel.api.SubnetAwareLocations
+import com.netflix.spinnaker.keel.core.api.ClusterDependencies
+
+/**
+ * A [ResourceSpec] that has default and override cluster dependencies (i.e. security groups, load
+ * balancers, and/or target groups).
+ */
+interface OverrideableClusterDependencyContainer<T : ClusterDependencyContainer> :
+  Locatable<SubnetAwareLocations> {
+  val defaults: T
+  val overrides: Map<String, T>
+}
+
+interface ClusterDependencyContainer {
+  val dependencies: ClusterDependencies?
+}
+
+/**
+ * Resolves overrides and returns a map of security group name to the regions it is required in.
+ * For example:
+ * ```
+ * fnord -> (us-east-1, us-west-2)
+ * fnord-elb -> (us-east-1, us-west-2)
+ * fnord-ext -> (us-east-1)
+ * ```
+ */
+val OverrideableClusterDependencyContainer<*>.securityGroupsByRegion: Map<String, Set<String>>
+  get() = dependencyByRegion { it.securityGroupNames }
+
+val OverrideableClusterDependencyContainer<*>.loadBalancersByRegion: Map<String, Set<String>>
+  get() = dependencyByRegion { it.loadBalancerNames }
+
+val OverrideableClusterDependencyContainer<*>.targetGroupsByRegion: Map<String, Set<String>>
+  get() = dependencyByRegion { it.targetGroups }
+
+private fun OverrideableClusterDependencyContainer<*>.dependencyByRegion(fn: (ClusterDependencies) -> Set<String>): Map<String, Set<String>> {
+  val regions = locations.regions.map { it.name }.toSet()
+  val defaults = (defaults.dependencies?.let(fn) ?: emptySet())
+    .associateWith { regions }
+  val overrides = overrides
+    .map { (region, spec) ->
+      spec.dependencies?.let(fn)?.associateWith { setOf(region) } ?: emptyMap()
+    }
+    .zip()
+  return listOf(defaults, overrides)
+    .zip()
+}
+
+/**
+ * Reduces a collection of `Map<*, Set>` down to a single map where the values contain the combined
+ * set of values from all elements in the original collection with a common key.
+ */
+private fun <K, VE> Collection<Map<K, Set<VE>>>.zip(): Map<K, Set<VE>> =
+  reduceOrDefault(emptyMap()) { acc, map ->
```

Comment:
what does `acc` stands for?

Reply:
My guess is `accumulator`. It's a common name for the first argument of a `reduce` or `fold` function.

Reply:
Accumulator. I think I just inherited that from the Kotlin method I'm implementing